tests/test_cli.py

```diff
@@ -65,6 +66,86 @@ def test_server_static_prefix_validation():
         run_server_mock.assert_not_called()
 
 
+def test_server_uvicorn_options():
+    """Test that uvicorn options are properly handled."""
+    with mock.patch("mlflow.server._run_server") as run_server_mock:
+        # Test default behavior (uvicorn should be used when no server options specified)
+        CliRunner().invoke(server)
+        run_server_mock.assert_called_once()
+        # Verify uvicorn is used by default
+        # _run_server parameters: file_store_path, registry_store_uri, default_artifact_root,
+        # serve_artifacts, artifacts_only, artifacts_destination, host, port, static_prefix,
+        # workers, gunicorn_opts, waitress_opts, expose_prometheus, app_name, uvicorn_opts
+        args = run_server_mock.call_args[0]
+        gunicorn_opts = args[10]
+        waitress_opts = args[11]
+        uvicorn_opts = args[14]
+        # uvicorn is used when gunicorn and waitress are not specified
+        assert gunicorn_opts is None
+        assert waitress_opts is None
+        assert uvicorn_opts is None
+
+    with mock.patch("mlflow.server._run_server") as run_server_mock:
+        # Test with uvicorn-opts
+        CliRunner().invoke(server, ["--uvicorn-opts", "--reload --log-level debug"])
+        run_server_mock.assert_called_once()
+        args = run_server_mock.call_args[0]
+        gunicorn_opts = args[10]
+        waitress_opts = args[11]
+        uvicorn_opts = args[14]
+        assert gunicorn_opts is None
+        assert waitress_opts is None
+        assert uvicorn_opts == "--reload --log-level debug"
+
+
+@pytest.mark.skipif(is_windows(), reason="Gunicorn is not supported on Windows")
+def test_server_gunicorn_options():
+    """Test that gunicorn options are properly handled."""
+    with mock.patch("mlflow.server._run_server") as run_server_mock:
+        # Test that gunicorn-opts disables uvicorn
+        CliRunner().invoke(server, ["--gunicorn-opts", "--log-level debug"])
+        run_server_mock.assert_called_once()
+        args = run_server_mock.call_args[0]
+        gunicorn_opts = args[10]
+        assert gunicorn_opts == "--log-level debug"
+
+    # Test conflicting options
+    result = CliRunner().invoke(
+        server, ["--uvicorn-opts", "--reload", "--gunicorn-opts", "--log-level debug"]
+    )
+    assert "Cannot specify multiple server options" in result.output
+
+
+@pytest.mark.asyncio
+async def test_mlflow_server_aiohttp():
+    """Test that the MLflow server works correctly with aiohttp client."""
+    port = get_safe_port()
+    cmd = ["mlflow", "server", "--port", str(port)]
+    process = subprocess.Popen(cmd)
+    try:
+        _await_server_up_or_die(port)
+
+        async with aiohttp.ClientSession() as session:
+            # Test basic endpoints work
```

Comment:
```suggestion
            # Test ASGI endpoints work
```

`basic` sounds a bit ambiguous.